prov/efa/src/efa_fabric.c

```diff
@@ -479,13 +489,30 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	memset(gdr_support_str, 0, sizeof(gdr_support_str));
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, sizeof(gdr_support_str));
+	if (ret < 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Cannot read sysfs file %s!\n", gdr_path);
+		goto out;
+	}
+
+	if (ret == 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Sysfs file %s is empty\n", gdr_path);
+		ret = -FI_EINVAL;
 		goto out;
+	}
 
-	ret = atoi(&gdr_support);
+	ret = 0;
```

Comment:
I'd move the ret = 0 after all the checks, for clarity reasons.

Reply:
Thanks! Done in the new revision.